wordpress-mobile/WordPress-iOS
# Aztec can now publish posts

Closes #6553 

# Summary
This PR wires up the publishing actions to the services to perform the actual actions. There are several things that are **TODOs for another PR**:
1. Removal of failed media from the post upon publish (after asking). #6635
2. Taking into account if a user can publish to the site or not (to implement "Submit for Review"). #6636
3. Dismissing of window is wonky - post-post is getting dismissed when it shouldn't be. #6637
4. Switching to post lists after publishing, if appropriate. (I think this is being handled by the editpost VC that handles post-post but I could be wrong.) #6637

# To test:
Make sure to test scenarios that include the following:
* Editing an existing post, creating a new post.
* Use the secondary publish button (publish now or save draft) from the action controller.
* Attempt to publish while media is uploading.
* Anything else you can think of to break crap.
* Test editing a New Post > Switching the Site > Saving as draft. Verify that the app doesn't get locked.

Needs review: @diegoreymendez, @jleandroperez, @SergioEstevao
